bin/extract_to_arb.dart

```diff
@@ -20,64 +18,84 @@ import 'package:intl_translation/src/directory_utils.dart';
 import 'package:path/path.dart' as path;
 
 void main(List<String> args) {
-  String targetDir;
-  String outputFilename;
-  String sourcesListFile;
-  bool transformer;
-  var parser = ArgParser();
-  var extraction = MessageExtraction();
-  String locale;
+  String targetDir = '.';
+  String outputFilename = 'intl_messages.arb';
+  String? sourcesListFile;
+  bool transformer = false;
+  ArgParser parser = ArgParser();
+  MessageExtraction extract = MessageExtraction();
+  String? locale;
+
+  /// Whether to include source_text in messages
+  bool includeSourceText = false;
+
+  /// If this is true, no translation meta data is written
+  bool suppressMetaData = false;
+
+  /// If this is true, the @@last_modified entry is not output.
+  bool suppressLastModified = false;
+
+  /// If this is true, then treat all warnings as errors.
+  bool warningsAreErrors = false;
   parser.addFlag('suppress-last-modified',
-      defaultsTo: false,
-      callback: (x) => extraction.suppressLastModified = x,
+      callback: (x) => suppressLastModified = x,
       help: 'Suppress @@last_modified entry.');
   parser.addFlag('suppress-warnings',
       defaultsTo: false,
-      callback: (x) => extraction.suppressWarnings = x,
+      callback: (x) => extract = extract.copyWith(suppressWarnings: x),
       help: 'Suppress printing of warnings.');
   parser.addFlag('suppress-meta-data',
-      defaultsTo: false,
-      callback: (x) => extraction.suppressMetaData = x,
+      callback: (x) => suppressMetaData = x,
       help: 'Suppress writing meta information');
   parser.addFlag('warnings-are-errors',
-      defaultsTo: false,
-      callback: (x) => extraction.warningsAreErrors = x,
+      callback: (x) => warningsAreErrors = x,
       help: 'Treat all warnings as errors, stop processing ');
   parser.addFlag('embedded-plurals',
       defaultsTo: true,
-      callback: (x) => extraction.allowEmbeddedPluralsAndGenders = x,
+      callback: (x) =>
+          extract = extract.copyWith(allowEmbeddedPluralsAndGenders: x),
       help: 'Allow plurals and genders to be embedded as part of a larger '
           'string, otherwise they must be at the top level.');
-  parser.addFlag('transformer',
-      defaultsTo: false,
+  parser.addFlag('transformer', //TODO: can this be removed?
```

Comment:
I'd add your github id to the todo to help future archaeologists - `// TODO(username): ...`.